### ISSUE TITLE
Test suite dependencies

### ISSUE THREAD

**Issue by edoddridge** [edoddridge/MIM#33]
Having two of the tests fail because of a missing dependency is suboptimal.

We should definitely keep valgrind in the test suite, but perhaps we don't set it to run every time.

Alternatively, we decide that the test suite is just for developers and Travis, so having some dependencies is not an issue.

**Comment by axch** [comment]
Yes, I think at this stage the thing to do is to have the test suite be developer-facing, and possibly add user-facing self-tests in the future.  When it comes into being, the user-facing suite should indeed not depend on valgrind :)
